src/test/java/com/thebois/models/beings/BeingTests.java

```diff
@@ -445,84 +440,117 @@ public void allStructuresCompletedNewPlaced() {
     }
 
     @Test
-    public void deliverItemsToCloseStructure() {
+    public void pawnDoNotTryToFindNewClosestStructureWhenStillIncomplete() {
         // Arrange
-        final Position positionA = new Position(10, 0);
-        final Position startPosition = new Position(11, 0);
+        final IPathFinder mockPathFinder = Mockito.mock(IPathFinder.class);
+        final IFinder mockFinder = Mockito.mock(IFinder.class);
 
-        final World world = new World(50);
-        final IPathFinder pathFinder = new AstarPathFinder(world);
+        final IStructure mockStructure = Mockito.mock(IStructure.class);
 
-        final Random mockRandom = Mockito.mock(Random.class);
-        when(mockRandom.nextInt(anyInt())).thenReturn(0);
-        when(mockRandom.nextInt(anyInt())).thenReturn(0);
+        final Position expectedPosition = new Position(10f, 20f);
 
-        world.createStructure(positionA);
+        when(mockFinder.findNearestIncompleteStructure(any()))
+            .thenReturn(Optional.of(mockStructure));
 
-        final IStructure structureAtA = world.getStructureAt(positionA).get();
+        when(mockStructure.getPosition()).thenReturn(expectedPosition);
 
-        final Pawn testPawn = new Pawn(startPosition,
-                                       new Position(),
-                                       mockRandom,
-                                       pathFinder,
-                                       world);
+        final Pawn pawn = new Pawn(new Position(),
+                                   new Position(),
+                                   new Random(),
+                                   mockPathFinder,
+                                   mockFinder);
 
         // Act
-        final float initialBuiltRatio = structureAtA.builtStatus();
-        testPawn.update();
+        pawn.update();
+        pawn.update();
 
         // Assert
-        assertThat(initialBuiltRatio).isLessThan(structureAtA.builtStatus());
+        final int callsExpected = 1;
+        verify(mockFinder, times(callsExpected)).findNearestIncompleteStructure(any());
     }
 
     @Test
-    public void findNewUnBuiltStructure() {
+    public void pawnTriesTpFindNewClosestStructureWhenNearestIsCompleted() {
         // Arrange
-        final Position positionA = new Position(10, 0);
-        final Position positionB = new Position(20, 0);
-        final Position startPosition = new Position(11, 0);
-
-        final World world = new World(50);
-        final IPathFinder pathFinder = new AstarPathFinder(world);
+        final IPathFinder mockPathFinder = Mockito.mock(IPathFinder.class);
+        final IFinder mockFinder = Mockito.mock(IFinder.class);
 
-        final Random mockRandom = Mockito.mock(Random.class);
-        when(mockRandom.nextInt(anyInt())).thenReturn(0);
+        final IStructure mockStructureA = Mockito.mock(IStructure.class);
+        final IStructure mockStructureB = Mockito.mock(IStructure.class);
```

Comment:
style

Instead of A and B, more meaningful names are `finishedStructure` and `unfinishedStructure`.

Reply:
Alright, my suggested names don't fit then, but my point still stands. `xxxxxA` and `xxxxxB` are very non-meaningful names, which makes it harder to read the code and fully understand what is going on.

Since you know the code the best out of the two of us, you should be able to come up with better names for them.
Maybe something like `structureUnderConstruction` and `unfinishedStructure`?